docs/madr/decisions/011-mesh-traffic-route.md

```diff
@@ -0,0 +1,555 @@
+the top, rule-independent level. This MADR proposes load balancing as a _filter_.
+This way it can be applied on a _per-route_ basis as well as a _per-backend_ basis.
+
+```yaml
+spec:
+  default:
+    rules:
+    - filters:
+      - type: LoadBalancing
+        loadBalancing: 
+          localityAware: true
+          type: RoundRobin|LeastRequest|RingHash|Random|Maglev
+          leastRequest:
+            choiceCount: 8
+          ringHash:
+            hashFunction: "MURMUR_HASH_2"
+            minRingSize: 64
+            maxRingSize: 1024
+            hashPolicies:
+              - # oneof
+                header:
+                  name:
+                  regexRewrite:
+                cookie:
+                  name:
+                  ttl:
+                  path:
+                connectionProperties:
+                  sourceIP: <bool>
+                queryParameter:
+                  name:
+                filterState:
+                  key:
+                terminal: <bool>
+          maglev:
+            tableSize: 65537
+            hashPolicies: ... # same schema as above
+```
+
+##### Gateway API
+
+There is no other extension point for `HTTPRoute` than `filters` so any additional configuration
+must appear under `filters`.
+
+With [`ExtensionRef`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilterType)
+a filter is powerful, flexible and extensible enough to support any arbitrary routing configuration.
+It doesn't really assume anything about the referenced resource.
+
+For example, to use round robin for all requests:
+
+```yaml
+spec:
+  ...
+  rules:
+    - matches: []
+      filters:
+        - type: ExtensionRef
+          extensionRef:
+            group: kuma.io/v1alpha1
+            kind: RouteConfig
+            name: load-balancer-common
+            # potential inlined load-balancer-common:
+            # spec:
+            #   loadBalancer: RoundRobin
+      backendRefs: []
+```
+
+The Gateway API controller would handle converting these resources to
+`MeshHTTPRoute`.
+
+#### Merging behavior
+
+With `targetRef` resources, the ultimate configuration is a result of overriden/merged policies depending on their specificity.
+
+> :question: Is the current merge behavior what we need for routing policy? What surprises are there? See the next sections
+
+##### Service owner routes
+
+There's a discussion in the Gateway API repository about what different types of
+routes exist and how they should interact. Consider:
+
+- the owner of a service `svc_a`
+may want to configure a routing policy for their service that applies to requests from all services.
+- another service owner may want to apply routing to requests from their service
+to `svc_a`.
+
+How should these routes be merged? One invariant might be that the consumer of a
+service shouldn't be able to override the routes from the service owner.
```

Comment:
> consumer of a service shouldn't be able to override the routes from the service owner

I'm not really convinced by this. Can you share an example case where this might be neccessary?

Reply:
I mean to me a good example is the service owner canarying `v2`. Without `override` or similar, any consumer can override the backends, unintentionally or otherwise.

Reply:
for the next MADR